docs/reference/balljoint.md

```diff
@@ -4,6 +4,14 @@ Derived from [Joint](joint.md).
 
 ```
 BallJoint {
```

Comment:
Why not document it is derived from Hinge2Joint and remove the jointParameters2, device, device2, position, position2 field to keep only the *3 fields? We did something similar for the Hinge2Joint with respect to the HingeJoint.